packages/beacon-node/src/api/rest/base.ts

```diff
@@ -45,12 +45,12 @@ export class RestApiServer {
       ajv: {customOptions: {coerceTypes: "array"}},
       querystringParser: (str) =>
         qs.parse(str, {
-          // defaults to 20 but Beacon API spec allows max items of 30
-          arrayLimit: 30,
-          // array as comma-separated values must be supported to be OpenAPI spec compliant
+          // Array as comma-separated values must be supported to be OpenAPI spec compliant
           comma: true,
-          // default limit of 1000 seems unnecessarily high, let's reduce it a bit
-          parameterLimit: 100,
```

Comment:
Parameter limit silently cuts off exceeding values which is not ideal, using the default value of `1000` here should be fine and is only relevant if some sends array queries like this `id=3&id=4&id=5`.

The parameter limit is anyhow not what should be used to limit DoS vectors, instead we should rely on maximum header size of the http server which is 16KB.

Some downstream tooling such as rocketpool might require this limit to be increased as they send requests like this `eth/v1/beacon/states/head/validators?id=<like 1000 validator pubkeys>` which requires much higher max header size.

For now, this configurable using the `--max-http-header-size` CLI option, see [http.maxHeaderSize](https://nodejs.org/api/http.html#httpmaxheadersize)

Reply:
Lodestar now also has a CLI flag `--rest.headerLimit` to configure this
- https://github.com/ChainSafe/lodestar/pull/5594